File: Java/easy/409. Longest Palindrome/409_Longest_Palindrome.java
import java.util.HashMap;

public class Solution {
    public int longestPalindrome(String s) {
        HashMap<Character, Integer> countMap = new HashMap<>();
        for (char c : s.toCharArray()) {
            countMap.put(c, countMap.getOrDefault(c, 0) + 1);
        }
        int length = 0;
        boolean oddFound = false;
        for (int count : countMap.values()) {
            if (count % 2 == 0) {
                length += count;
            } else {
                length += count - 1;
                oddFound = true;
            }
        }
        return oddFound ? length + 1 : length;
    }
}

File: Golang/409. Longest Palindrome/409_Longest_Palindrome.go
func longestPalindrome(s string) int {
    charCount := make(map[rune]int)
    for _, ch := range s {
        charCount[ch]++
    }
    length := 0
    oddFound := false
    for _, count := range charCount {
        if count%2 == 0 {
            length += count
        } else {
            length += count - 1
            oddFound = true
        }
    }
    if oddFound {
        length++
    }
    return length
}
